### PR TITLE
feat: instant meeting improvements

### LINKED ISSUE #13174: [CAL-2913] Improvements in instant meetings
- [X] Display team members pfp
- [X]  Add option to add webhook in Switch
- [X] a countdown of 60 seconds: "We’ll connect you within 60 seconds or find a date slot in the future in 45, 44, 43 etc.”
- [X] Afterwards: "Oops, we couldn't connect you this time. Please schedule a future call instead"
- [ ] royalty free lo-fi music
- [X] a mini game while waitiing? flappy birds or chrome dinosaur run?
- [X] lets block "closing tab" while waiting, i think there is a browser API to throw an alert when closing a tab
- [ ] lets play a sound when the "wait screen" turns into the video link and someone joins (some people change tab while waiting)

new:

> We are connecting you!
> Please schedule a future call if we're not available in XX seconds.

Then...

> Oops, we couldn't connect you this time.
> Please schedule a future call instead. We value your time.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2913](https://linear.app/calcom/issue/CAL-2913/improvements-in-instant-meetings)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/13174

-  Display team members pfp
- a countdown of 60 seconds
- Afterwards: "Oops, we couldn't connect you this time. Please schedule a future call instead"
- lets block "closing tab" while waiting
- Add option to add webhook in Switch

1) 
Screenshot 2024-01-16 at 8 23 57 PM

Webhooks

https://github.com/calcom/cal.com/assets/53316345/de6ca1ac-0e17-4b52-af75-3fd03cf9e712

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: apps/web/components/eventtype/InstantEventController.tsx [hunk 1]
```diff
@@ -97,3 +99,193 @@ export default function InstantEventController({
     </LicenseRequired>
   );
 }
+
+const InstantMeetingWebhooks = ({ eventType }: { eventType: EventTypeSetup }) => {
+  const { t } = useLocale();
+  const utils = trpc.useContext();
+  const { data: webhooks } = trpc.viewer.webhook.list.useQuery({ eventTypeId: eventType.id });
```
Comment (CarinaWolli) [inline apps/web/components/eventtype/InstantEventController.tsx:106]: Why do we want to list all webhooks in the Instant Booking tab?

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
The 60s countdown doesn't work for me, it always shows 0s until I am getting the 'Oops, ... message. @Udit-takkar is that working for you?

Also, that looks like a lot of text, really close together. Can we make that prettier? 
Screenshot 2024-01-17 at 5 05 51 PM

File: apps/web/components/eventtype/InstantEventController.tsx [hunk 2]
```diff
@@ -97,3 +99,193 @@ export default function InstantEventController({
     </LicenseRequired>
   );
 }
+
+const InstantMeetingWebhooks = ({ eventType }: { eventType: EventTypeSetup }) => {
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply apps/web/components/eventtype/InstantEventController.tsx:103]: I tried but the design is a bit different then we need

File: apps/web/components/eventtype/InstantEventController.tsx (hunk 1 above)
Comment (Udit-takkar) [reply apps/web/components/eventtype/InstantEventController.tsx:106]: Oh right. forgot to update this

File: apps/web/components/eventtype/InstantEventController.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply apps/web/components/eventtype/InstantEventController.tsx:104]: We should refactor the code then so we can reuse as much as possible, looks like a lot of duplicate code. But we don't need to do that in this PR

File: packages/features/instant-meeting/handleInstantMeeting.ts [hunk 3]
```diff
@@ -179,7 +179,8 @@ async function handler(req: NextApiRequest) {
   const instantMeetingToken = await prisma.instantMeetingToken.create({
     data: {
       token,
-      expires: new Date(new Date().getTime() + 1000 * 60 * 5),
+      // 90 Seconds
+      expires: new Date(new Date().getTime() + 1000 * 90),
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply packages/features/instant-meeting/handleInstantMeeting.ts:183]: asked because in the PR description it says 'countdown of 60s'

**Review by CarinaWolli — DISMISSED** [review]
Code looks good and works 👍

**Review by zomars — DISMISSED** [review]

**Review by PeerRich — DISMISSED** [review]
awesome!

File: packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx [hunk 4]
```diff
@@ -557,19 +601,16 @@ const RedirectToInstantMeetingModal = ({ expiryTime }: { expiryTime?: Date }) =>
...
+                {t("please_schedule_future_call", {
+                  seconds: formatTime(timeRemaining),
+                })}
+              </p>
 
-              <p className="font-medium">{t("please_do_not_close_this_tab")}</p>
-              <div className="h-[414px]">
-                <iframe className="mx-auto h-full w-[276px] rounded-lg" src="https://cal.games/" />
+              <div className="mt-4 h-[414px]">
+                <iframe className="mx-auto h-full" src="https://cal.games/" />
```
Comment (PeerRich) [inline packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx:613]:
```suggestion
                <iframe className="mx-auto h-full w-[276px] rounded-lg"" src="https://cal.games/" />
```

**Review by PeerRich — DISMISSED** [review]

File: apps/web/public/static/locales/en/common.json [hunk 5]
```diff
@@ -112,7 +112,7 @@
   "requested_to_reschedule_subject_attendee": "Action Required Reschedule: Please book a new time for {{eventType}} with {{name}}",
   "hi_user_name": "Hi {{name}}",
   "ics_event_title": "{{eventType}} with {{name}}",
-  "please_book_a_time_sometime_later":"No one was available at the moment. Please book for sometime later",
+  "please_book_a_time_sometime_later":"Oops, we couldn't connect you this time. Please schedule a future call instead",
```
Comment (PeerRich) [inline apps/web/public/static/locales/en/common.json:115]:
```suggestion
  "please_book_a_time_sometime_later":"Oops, we couldn't connect you this time. Please schedule a call in the future instead",
```

**Review by PeerRich — DISMISSED** [review]

**Comment by CarinaWolli** [comment]
Type-check is failing now and there are merge conflicts, can you take a look @Udit-takkar?

**Review by CarinaWolli — APPROVED** [review]
Great work @Udit-takkar 👏🏻

**Pull request merged**